lib/galaxy/webapps/galaxy/api/dataset_collections.py

```diff
@@ -64,6 +69,48 @@ def create(self, trans: ProvidesHistoryContext, payload: dict, **kwd):
         return dictify_dataset_collection_instance(dataset_collection_instance,
                                                    security=trans.security, parent=create_params["parent"])
 
+    @expose_api
+    def update(self, trans: ProvidesHistoryContext, payload: dict, id):
+        """
+        Iterate over all datasets of a collection and copy datasets with new attributes to a new collection.
+        e.g attributes = {'dbkey': 'dm3'}
+
+        * PUT /api/dataset_collections/{hdca_id}:
+            create a new dataset collection instance.
+        """
+
+        if len(payload) != 1:
+            raise Exception("Update one attribute at a time.")
+        if 'dbkey' not in payload:
+            raise Exception("This attribute cannot be modified.")
```

Comment:
That's in direct contradiction to the docstring, but didn't you also want to allow changing the extension?

In either case you want to raise an exception that is being caught and forwarded in the http response
```suggestion
        if len(payload) != 1:
            raise exceptions.RequestParameterInvalidException("Update one attribute at a time.")
        if 'dbkey' not in payload:
            raise exceptions.RequestParameterInvalidException("This attribute cannot be modified.")
```